lib/Backends/CPU/CPUFunction.h

```diff
@@ -27,14 +27,14 @@ class CPUFunction final : public CompiledFunction {
   /// The LLVM JIT engine. The jit must be initialized after the ctor
   /// initializes the LLVM backends.
   std::unique_ptr<llvm::orc::GlowJIT> JIT_;
-  /// runtimeBundle contains symbol offsets and allocation sizes.
-  runtime::RuntimeBundle runtimeBundle_;
   /// Base address for Activations memory block.
   uint8_t *baseActivationsAddress_{};
   /// Base address for Mutable weights memory block, Inputs and Outputs.
   uint8_t *baseMutableWeightVarsAddress_{};
 
 public:
+  /// runtimeBundle contains symbol offsets and allocation sizes.
+  runtime::RuntimeBundle runtimeBundle_;
```

Comment:
I recommend keeping runtimeBundle private; having public data members has a way of bloating the API.  It seems like you should be able to just add a `collectConstants` method to `CPUFunction`.  (Or maybe even to the `CompiledFunction` interface, since it's used by every backend.)

Reply:
> Or maybe even to the CompiledFunction interface, since it's used by every backend.

this way we'd also avoid ugly static_casts on pointers to specific CompiledFunctions.

Reply:
I think adding it as a method to each compiledFunction makes sense, I think eventually this will go away entirely, the new DeviceManager will have the collection logic. Unfortunately even if I added it to CompiledFunction OpenCL and Interpreter will still require a static_cast to get the IR. 

So how about I keep bundle private and add a new method to each CPUFunction, OpenCLFunction, and InterpreterFunction but not to compiledFunction, since ultimately we'd want to move this to the DeviceManager anyway?